sync/sync.go

```diff
@@ -122,7 +133,10 @@ func (s *Synchronizer) syncBlocks(syncCtx context.Context) {
 	verifiers := stream.New().WithMaxGoroutines(runtime.NumCPU())
 
 	streamCtx, streamCancel := context.WithCancel(syncCtx)
+
 	nextHeight := s.nextHeight()
+	startingHeight := nextHeight
+	s.StartingBlockNumber = &startingHeight
```

Comment:
any chance we do this with out the `startingHeight`?
also, dont we need to set `s.StartingBlockNumber` to nil when exiting `syncBlocks`?

Reply:
Got it now. Thanks

Reply:
Done

Reply:
`s.HighestBlockHeader` needs to be cleared as well. lets do both of those in a defer statement at the beginning of the `syncBlocks`

Reply:
updated it